example/parquet_reader.cc

```diff
@@ -153,43 +156,46 @@ void* read_parquet(char* filename) {
           break;
       }
 
+      bool is_null;
       int def_level = 0, rep_level = 0;
       while (reader.HasNext()) {
         switch (col.meta_data.type) {
           case Type::BOOLEAN: {
-            bool val = reader.GetBool(&def_level, &rep_level);
-            if (def_level < rep_level) break;
-            ((bool*)(((bool**)column_ptr)[c]))[num_values] = val;
+            bool val = reader.GetBool(&is_null, &def_level, &rep_level);
+            ((bool*)(column_ptr[c]))[num_values] = val;
             break;
           }
           case Type::INT32: {
-            int32_t val = reader.GetInt32(&def_level, &rep_level);;
-            if (def_level < rep_level) break;
-            ((int32_t*)(((int32_t**)column_ptr)[c]))[num_values] = val;
+            int32_t val = reader.GetInt32(&is_null, &def_level, &rep_level);
+            ((int32_t*)(column_ptr[c]))[num_values] = val;
             break;
           }
           case Type::INT64: {
-            int64_t val = reader.GetInt64(&def_level, &rep_level);;
-            if (def_level < rep_level) break;
-            ((int64_t *)(((int64_t**)column_ptr)[c]))[num_values] = val;
+            int64_t val = reader.GetInt64(&is_null, &def_level, &rep_level);
+            ((int64_t*)(column_ptr[c]))[num_values] = val;
             break;
           }
           case Type::FLOAT: {
-            float val = reader.GetFloat(&def_level, &rep_level);;
-            if (def_level < rep_level) break;
-            ((float*)(((float**)column_ptr)[c]))[num_values] = val;
+            float val = reader.GetFloat(&is_null, &def_level, &rep_level);
+            ((float*)(column_ptr[c]))[num_values] = val;
             break;
           }
           case Type::DOUBLE: {
-            double val = reader.GetDouble(&def_level, &rep_level);;
-            if (def_level < rep_level) break;
-            ((double*)(((double**)column_ptr)[c]))[num_values] = val;
+            double val = reader.GetDouble(&is_null, &def_level, &rep_level);
+            ((double*)(column_ptr[c]))[num_values] = val;
             break;
           }
           case Type::BYTE_ARRAY: {
-            ByteArray val = reader.GetByteArray(&def_level, &rep_level);;
-            if (def_level < rep_level) break;
-            ((ByteArray*)(((ByteArray**)column_ptr)[c]))[num_values] = val;
+            ByteArray val = reader.GetByteArray(&is_null, &def_level, &rep_level);
+            ByteArray* dst = &((ByteArray*)(column_ptr[c]))[num_values];
+            if (is_null) {
+              dst->ptr = NULL;
+              dst->len = 0;
+            } else {
+              dst->len = val.len;
+              dst->ptr = new uint8_t[val.len];
+              memcpy((char*)dst->ptr, val.ptr, val.len);
+            }
```

Comment:
why the if (is_null) only for bytearray? should this logic (dst->ptr = Null ...) be in reader.GetByteArray ?